juriscraper/opinions/united_states/state/nh.py

```diff
@@ -1,127 +1,58 @@
-
-    def _get_docket_numbers(self):
-        return [pair["docket"] for pair in self.docket_name_pairs]
-
-    def _get_anchor_docket_name_pairs(self):
-        """The court has some ugly HTML practices that we need to handle.
-
-        Most anchor links include single line strings with a single docket
-        number and a single case name.  However, there are two other formats
-        we've seen and must work around.
-
-        (CASE 1)
-        The anchor has multiple lines broken with <br> tag(s), and each
-        line contains "<docket> <name>". In this case we need to combine
-        the docket numbers and name strings respectively.
-        [EXAMPLE: February 18, 2016 in nh_example_2.html]
-
-        (CASE 2)
-        The anchor has multiple lines broken with <br> tag(s), and the
-        second line is a continuation of a long case name started on the first
-        line.  So, the second line does not lead with a docket number, thus
-        this line's string should be glued onto the <name> substring extracted
-        from the previous line.
-        [EXAMPLE: September 18, 2018 in nh_example_6.html]
-        """
-        pairs = []
-        for anchor in self.html.xpath(self.link_path):
-            i = 0
-            dockets = []
-            name_substrings = []
-            text_anchor = anchor.text_content()
-            text_clean = text_anchor.replace("\n", "")
-
-            for text in text_clean.split(";"):
-                text = clean_string(text)
-                match = self.link_text_regex.search(text)
-                try:
-                    docket = match.group(1)
-                    dockets.append(docket)
-                    name = match.group(2)
-                    name = " ".join(name.split())
-                    name_substrings.append(name)
-                    i += 1
-                except AttributeError:
-                    if i == 0:
-                        # docket and name (root) should be contained in first substring
-                        error = f"Invalid anchor root string format: {text}"
-                        raise InsanityException(error)
-                    # no docket in the substring, its a trailing name substring
-                    # that they broke over multiple lines, so glue it to the
-                    # previous name substring
-                    name_substrings[i - 1] += f" {text}"
-            pairs.append(
+        self.url = "https://www.courts.nh.gov/content/api/documents"
+        self.method = "GET"
+        self.parameters = {
+            "sort": "field_date_posted|desc",
+            "page": "1",
+            "size": "25",
+            "purpose": "1331+undefined",
+            "tag": "1206+",
+        }
+
+    def _download(self, request_dict={}):
+        if self.test_mode_enabled():
+            return super()._download(request_dict)
+        resp = self.request["session"].get(self.url, params=self.parameters)
+        return resp.json()
```

Comment:
@mlissner --- can I get a second opinion on this.  Is this a fine way to use _download.  I suggested it to @satsuki-chan earlier but I would like your thoughts.

Reply:
I have the same question @mlissner  👆
It was the only way I found to load the example file to run the test for this scraper.

Reply:
TL;DR: Let's make ourselves happy the next time we are dealing with JSON.

------

I think the big picture here is that we don't have a good way to deal with JSON? If that assumption is correct, yeah, the above would *work*, but I think this is a common-enough use case that instead of working around our own code with hacks like this, we should add functionality to handle this in an elegant way. 

I think in that case, that could mean a change to the `_download` function itself or maybe something else in the AbstractSite. I haven't looked closely enough to know the right solution.

It pains me to see this function abused like this to work around our own design assumptions (that there'd never be JSON coming from courts). Instead of hacking around our own code base, @satsuki-chan would you mind proposing a solution that's actually good?

Reply:
Thanks @mlissner

Reply:
@mlissner , the current architecture _**is able to process JSON responses**_ in method `_return_response_text_object`:
https://github.com/freelawproject/juriscraper/blob/23e61f8a391c9d3c68b9194168c9ee10e6a7ae85/juriscraper/AbstractSite.py#L382-L388

What is missing is that the optional parameters for page requests in dictionary `self.parameters` are _**only included when the requests is made with POST**_ method:
https://github.com/freelawproject/juriscraper/blob/23e61f8a391c9d3c68b9194168c9ee10e6a7ae85/juriscraper/AbstractSite.py#L359-L369

To solve this kind of issues, is necessary to also add the parameter `data` in the **GET** request in method `_request_url_get`:
https://github.com/freelawproject/juriscraper/blob/23e61f8a391c9d3c68b9194168c9ee10e6a7ae85/juriscraper/AbstractSite.py#L346-L357

```
def _request_url_get(self, url):
    """Execute GET request and assign appropriate request dictionary
    values
    """
    self.request["url"] = url
    self.request["response"] = self.request["session"].get(
        url,
        headers=self.request["headers"],
        verify=self.request["verify"],
        data=self.parameters,
        timeout=60,
        **self.request["parameters"],
    )
```

Given the scope and efect that his change might have in all other scrapers, I propose to make this change in a separate PR with fixes for all scrapers that report issues with this change in their GET requests.

Reply:
To clarify, POST requests to the New Hampshire API are not allowed. In my tests with the POST methods I got the HTTP code **405**, Method not allowed.

Reply:
I see. If that's all that's needed, I'm fine with adding that in a separate PR, but in the meantime, why not just add the GET parameters to the URL?

Reply:
Done.